Softvector Build Fails on Mac and Windows

The [softvector build](https://github.com/tum-ei-eda/etiss/pull/76) on Mac, more specifically the linking, fails for some reason. The build for Linux appears to work just fine (until it gets stopped by the failing Mac build). The linker is having a hard time finding some symbols:

```
[ 91%] Linking CXX shared library libetiss_softvector.dylib
Undefined symbols for architecture x86_64:
  "VLSU::load_eew(std::__1::function<void (unsigned long, unsigned char*, unsigned long)>, unsigned char*, unsigned long long, unsigned long long, unsigned short, unsigned short, unsigned short, unsigned short, unsigned long long, unsigned short, unsigned char, short)", referenced from:
      _etiss_vload_encoded_unitstride in etiss_softvector.cpp.o
      _etiss_vload_encoded_stride in etiss_softvector.cpp.o
      _etiss_vload_segment_unitstride in etiss_softvector.cpp.o
      _etiss_vload_segment_stride in etiss_softvector.cpp.o
  "VLSU::store_eew(std::__1::function<void (unsigned long, unsigned char*, unsigned long)>, unsigned char*, unsigned long long, unsigned long long, unsigned short, unsigned short, unsigned short, unsigned short, unsigned long long, unsigned short, unsigned char, short)", referenced from:
      _etiss_vstore_encoded_unitstride in etiss_softvector.cpp.o
      _etiss_vstore_encoded_stride in etiss_softvector.cpp.o
      _etiss_vstore_segment_unitstride in etiss_softvector.cpp.o
      _etiss_vstore_segment_stride in etiss_softvector.cpp.o
  "VTYPE::decode(unsigned short, unsigned char*, unsigned char*, unsigned int*, unsigned char*, unsigned char*)", referenced from:
      _etiss_vload_encoded_unitstride in etiss_softvector.cpp.o
      _etiss_vload_encoded_stride in etiss_softvector.cpp.o
      _etiss_vload_segment_unitstride in etiss_softvector.cpp.o
      _etiss_vload_segment_stride in etiss_softvector.cpp.o
      _etiss_vstore_encoded_unitstride in etiss_softvector.cpp.o
      _etiss_vstore_encoded_stride in etiss_softvector.cpp.o
      _etiss_vstore_segment_unitstride in etiss_softvector.cpp.o
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/jitlibs/softvector/libetiss_softvector.dylib] Error 1
make[1]: *** [src/jitlibs/softvector/CMakeFiles/etiss_softvector.dir/all] Error 2
make: *** [all] Error 2
Error: Process completed with exit code 2.
``` 

EDIT: The windows build appears to be failing as well (sorry for the confusion).

--- a/include/target/riscv/base/softvector-platform-types.hpp
+++ b/include/target/riscv/base/softvector-platform-types.hpp
@@ -60,9 +60,9 @@ public:
 	/// \param v RVVector to test against
 	/// \returns 1 if v starts within this. -1 if this starts within v. 0 if no overlap detected.
 	int8_t check_mem_overlap(const RVVector& v) {
-		if( (v.mem_ >= mem_) and (v.mem_ < mem_+length_*activeElement.width_in_bits_))
+		if( (v.mem_ >= mem_) && (v.mem_ < mem_+length_*activeElement.width_in_bits_))
 			return 1;
-		if( (mem_ >= v.mem_) and (mem_ < v.mem_+v.length_*v.activeElement.width_in_bits_))
+		if( (mem_ >= v.mem_) && (mem_ < v.mem_+v.length_*v.activeElement.width_in_bits_))
 			return -1;
 		return 0;
 	}

--- a/src/target/riscv/lsu/lsu.cpp
+++ b/src/target/riscv/lsu/lsu.cpp
@@ -48,7 +48,7 @@ VILL::vpu_return_t VLSU::load_eew(
 		RVVector& vd = V.get_vec(dst_vec_reg);
 		int memOffset = src_mem_start;
 		for(size_t iElement = 0; iElement < vec_len; ++iElement) {
-			if(iElement >= vec_elem_start and (mask_f or V.get_mask_reg().get_bit(iElement))) {
+			if(iElement >= vec_elem_start && (mask_f || V.get_mask_reg().get_bit(iElement))) {
 				func_read_mem(memOffset , vd[iElement].mem_, eew_bytes);
 			}
 			memOffset += (eew_bytes + stride_bytes);
@@ -82,7 +82,7 @@ VILL::vpu_return_t VLSU::store_eew(
 		RVVector& vs3 = V.get_vec(src_vec_reg);
 		int memOffset = dst_mem_start;
 		for(size_t iElement = 0; iElement < vec_len; ++iElement) {
-			if(iElement >= vec_elem_start and (mask_f or V.get_mask_reg().get_bit(iElement))) {
+			if(iElement >= vec_elem_start && (mask_f || V.get_mask_reg().get_bit(iElement))) {
 				func_write_mem(memOffset , vs3[iElement].mem_, eew_bytes);
 			}
 			memOffset += (eew_bytes + stride_bytes);

--- a/src/vpu/softvector-register.cpp
+++ b/src/vpu/softvector-register.cpp
@@ -44,7 +44,7 @@ void SVRegister::m_assign(
 	size_t start_index
 ) {
 	for(size_t i_byte = start_index/8; i_byte < length_bits_; ++i_byte) {
-		if((i_byte == start_index/8) and (start_index != 0)) {
+		if((i_byte == start_index/8) && (start_index != 0)) {
 			for(size_t i_bit = start_index % 8; i_bit <8; ++i_bit) {
 				(*this)[i_byte] = ((*this)[i_byte] & (~vm[i_byte] & (0x1 << i_bit))) ^ (in[i_byte] & vm[i_byte]  & (0x1 << i_bit));
 			}
@@ -56,7 +56,7 @@ void SVRegister::m_assign(
 
 SVRegister& SVRegister::m_eq(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] == rhs[i_element]);
 	}
 	return(*this);
@@ -64,7 +64,7 @@ SVRegister& SVRegister::m_eq(const SVector& opL, const SVector& rhs,  const SVRe
 
 SVRegister& SVRegister::m_eq(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] == rhs);
 	}
 	return(*this);
@@ -72,7 +72,7 @@ SVRegister& SVRegister::m_eq(const SVector& opL, const int64_t rhs, const SVRegi
 
 SVRegister& SVRegister::m_neq(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] != rhs[i_element]);
 	}
 	return(*this);
@@ -80,7 +80,7 @@ SVRegister& SVRegister::m_neq(const SVector& opL, const SVector& rhs,  const SVR
 
 SVRegister& SVRegister::m_neq(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] != rhs);
 	}
 	return(*this);
@@ -88,7 +88,7 @@ SVRegister& SVRegister::m_neq(const SVector& opL, const int64_t rhs, const SVReg
 
 SVRegister& SVRegister::m_s_lt(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] < rhs[i_element]);
 	}
 	return(*this);
@@ -96,7 +96,7 @@ SVRegister& SVRegister::m_s_lt(const SVector& opL, const SVector& rhs,  const SV
 
 SVRegister& SVRegister::m_s_lt(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] < rhs);
 	}
 	return(*this);
@@ -104,7 +104,7 @@ SVRegister& SVRegister::m_s_lt(const SVector& opL, const int64_t rhs, const SVRe
 
 SVRegister& SVRegister::m_s_lte(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] <= rhs[i_element]);
 	}
 	return(*this);
@@ -112,7 +112,7 @@ SVRegister& SVRegister::m_s_lte(const SVector& opL, const SVector& rhs,  const S
 
 SVRegister& SVRegister::m_s_lte(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] <= rhs);
 	}
 	return(*this);
@@ -120,7 +120,7 @@ SVRegister& SVRegister::m_s_lte(const SVector& opL, const int64_t rhs, const SVR
 
 SVRegister& SVRegister::m_s_gt(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] > rhs[i_element]);
 	}
 	return(*this);
@@ -128,7 +128,7 @@ SVRegister& SVRegister::m_s_gt(const SVector& opL, const SVector& rhs,  const SV
 
 SVRegister& SVRegister::m_s_gt(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] > rhs);
 	}
 	return(*this);
@@ -136,7 +136,7 @@ SVRegister& SVRegister::m_s_gt(const SVector& opL, const int64_t rhs, const SVRe
 
 SVRegister& SVRegister::m_s_gte(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] >= rhs[i_element]);
 	}
 	return(*this);
@@ -144,7 +144,7 @@ SVRegister& SVRegister::m_s_gte(const SVector& opL, const SVector& rhs,  const S
 
 SVRegister& SVRegister::m_s_gte(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element] >= rhs);
 	}
 	return(*this);
@@ -153,7 +153,7 @@ SVRegister& SVRegister::m_s_gte(const SVector& opL, const int64_t rhs, const SVR
 // unsigned comparisons
 SVRegister& SVRegister::m_u_lt(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element].op_u_lt(rhs[i_element]));
 	}
 	return(*this);
@@ -161,7 +161,7 @@ SVRegister& SVRegister::m_u_lt(const SVector& opL, const SVector& rhs,  const SV
 
 SVRegister& SVRegister::m_u_lt(const SVector& opL, const uint64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element].op_u_lt(rhs));
 	}
 	return(*this);
@@ -169,7 +169,7 @@ SVRegister& SVRegister::m_u_lt(const SVector& opL, const uint64_t rhs, const SVR
 
 SVRegister& SVRegister::m_u_lte(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element].op_u_lte(rhs[i_element]));
 	}
 	return(*this);
@@ -177,7 +177,7 @@ SVRegister& SVRegister::m_u_lte(const SVector& opL, const SVector& rhs,  const S
 
 SVRegister& SVRegister::m_u_lte(const SVector& opL, const uint64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element].op_u_lte(rhs));
 	}
 	return(*this);
@@ -185,7 +185,7 @@ SVRegister& SVRegister::m_u_lte(const SVector& opL, const uint64_t rhs, const SV
 
 SVRegister& SVRegister::m_u_gt(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element].op_u_gt(rhs[i_element]));
 	}
 	return(*this);
@@ -193,7 +193,7 @@ SVRegister& SVRegister::m_u_gt(const SVector& opL, const SVector& rhs,  const SV
 
 SVRegister& SVRegister::m_u_gt(const SVector& opL, const uint64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element].op_u_gt(rhs));
 	}
 	return(*this);
@@ -201,7 +201,7 @@ SVRegister& SVRegister::m_u_gt(const SVector& opL, const uint64_t rhs, const SVR
 
 SVRegister& SVRegister::m_u_gte(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element].op_u_gte(rhs[i_element]));
 	}
 	return(*this);
@@ -209,7 +209,7 @@ SVRegister& SVRegister::m_u_gte(const SVector& opL, const SVector& rhs,  const S
 
 SVRegister& SVRegister::m_u_gte(const SVector& opL, const uint64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < opL.length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			set_bit(i_element, opL[i_element].op_u_gte(rhs));
 	}
 	return(*this);

--- a/src/vpu/softvector-vector.cpp
+++ b/src/vpu/softvector-vector.cpp
@@ -327,21 +327,21 @@ SVRegister SVector::op_u_gte(const uint64_t rhs) const {
 // masked ops
 void SVector::m_assign(const SVector& vin, const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element] = vin[i_element];
 	}
 }
 
 void SVector::m_assign(const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element] = rhs;
 	}
 }
 
 SVector& SVector::m_add(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_add(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -350,7 +350,7 @@ SVector& SVector::m_add(const SVector& opL, const SVector& rhs,  const SVRegiste
 
 SVector& SVector::m_add(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_add(opL[i_element], rhs);
 	}
 	return(*this);
@@ -358,7 +358,7 @@ SVector& SVector::m_add(const SVector& opL, const int64_t rhs, const SVRegister&
 
 SVector& SVector::m_sub(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_sub(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -366,7 +366,7 @@ SVector& SVector::m_sub(const SVector& opL, const SVector& rhs,  const SVRegiste
 
 SVector& SVector::m_sub(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_sub(opL[i_element], rhs);
 	}
 	return(*this);
@@ -374,7 +374,7 @@ SVector& SVector::m_sub(const SVector& opL, const int64_t rhs, const SVRegister&
 
 SVector& SVector::m_rsub(const int64_t lhs, const SVector& opR, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_rsub(lhs, opR[i_element]);
 	}
 	return(*this);
@@ -382,7 +382,7 @@ SVector& SVector::m_rsub(const int64_t lhs, const SVector& opR, const SVRegister
 
 SVector& SVector::m_waddu(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_waddu(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -390,7 +390,7 @@ SVector& SVector::m_waddu(const SVector& opL, const SVector& rhs,  const SVRegis
 
 SVector& SVector::m_waddu(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_waddu(opL[i_element], rhs);
 	}
 	return(*this);
@@ -398,7 +398,7 @@ SVector& SVector::m_waddu(const SVector& opL, const int64_t rhs, const SVRegiste
 
 SVector& SVector::m_wsubu(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_wsubu(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -406,7 +406,7 @@ SVector& SVector::m_wsubu(const SVector& opL, const SVector& rhs,  const SVRegis
 
 SVector& SVector::m_wsubu(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_wsubu(opL[i_element], rhs);
 	}
 	return(*this);
@@ -414,7 +414,7 @@ SVector& SVector::m_wsubu(const SVector& opL, const int64_t rhs, const SVRegiste
 
 SVector& SVector::m_wadd(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_wadd(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -422,7 +422,7 @@ SVector& SVector::m_wadd(const SVector& opL, const SVector& rhs,  const SVRegist
 
 SVector& SVector::m_wadd(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_wadd(opL[i_element], rhs);
 	}
 	return(*this);
@@ -430,7 +430,7 @@ SVector& SVector::m_wadd(const SVector& opL, const int64_t rhs, const SVRegister
 
 SVector& SVector::m_wsub(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_wsub(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -438,7 +438,7 @@ SVector& SVector::m_wsub(const SVector& opL, const SVector& rhs,  const SVRegist
 
 SVector& SVector::m_wsub(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_wsub(opL[i_element], rhs);
 	}
 	return(*this);
@@ -446,56 +446,56 @@ SVector& SVector::m_wsub(const SVector& opL, const int64_t rhs, const SVRegister
 //12.10 Multiplikation
 SVector& SVector::m_ssmul(const SVector& opL, const SVector& rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_ssmul(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
 }
 SVector& SVector::m_ssmul(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_ssmul(opL[i_element], rhs);
 	}
 	return(*this);
 }
 SVector& SVector::m_ssmulh(const SVector& opL, const SVector& rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_ssmulh(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
 }
 SVector& SVector::m_ssmulh(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_ssmulh(opL[i_element], rhs);
 	}
 	return(*this);
 }
 SVector& SVector::m_uumulh(const SVector& opL, const SVector& rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_uumulh(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
 }
 SVector& SVector::m_uumulh(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_uumulh(opL[i_element], rhs);
 	}
 	return(*this);
 }
 SVector& SVector::m_sumulh(const SVector& opL, const SVector& rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_sumulh(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
 }
 SVector& SVector::m_sumulh(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_sumulh(opL[i_element], rhs);
 	}
 	return(*this);
@@ -503,7 +503,7 @@ SVector& SVector::m_sumulh(const SVector& opL, const int64_t rhs, const SVRegist
 
 SVector& SVector::m_and(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_and(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -511,7 +511,7 @@ SVector& SVector::m_and(const SVector& opL, const SVector& rhs,  const SVRegiste
 
 SVector& SVector::m_and(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_and(opL[i_element], rhs);
 	}
 	return(*this);
@@ -519,7 +519,7 @@ SVector& SVector::m_and(const SVector& opL, const int64_t rhs, const SVRegister&
 
 SVector& SVector::m_or(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_or(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -527,7 +527,7 @@ SVector& SVector::m_or(const SVector& opL, const SVector& rhs,  const SVRegister
 
 SVector& SVector::m_or(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_or(opL[i_element], rhs);
 	}
 	return(*this);
@@ -535,7 +535,7 @@ SVector& SVector::m_or(const SVector& opL, const int64_t rhs, const SVRegister& 
 
 SVector& SVector::m_xor(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_xor(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -543,7 +543,7 @@ SVector& SVector::m_xor(const SVector& opL, const SVector& rhs,  const SVRegiste
 
 SVector& SVector::m_xor(const SVector& opL, const int64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_xor(opL[i_element], rhs);
 	}
 	return(*this);
@@ -551,7 +551,7 @@ SVector& SVector::m_xor(const SVector& opL, const int64_t rhs, const SVRegister&
 
 SVector& SVector::m_sll(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_sll(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -559,7 +559,7 @@ SVector& SVector::m_sll(const SVector& opL, const SVector& rhs,  const SVRegiste
 
 SVector& SVector::m_sll(const SVector& opL, const uint64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_sll(opL[i_element], rhs);
 	}
 	return(*this);
@@ -567,7 +567,7 @@ SVector& SVector::m_sll(const SVector& opL, const uint64_t rhs, const SVRegister
 
 SVector& SVector::m_sra(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_sra(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -575,7 +575,7 @@ SVector& SVector::m_sra(const SVector& opL, const SVector& rhs,  const SVRegiste
 
 SVector& SVector::m_sra(const SVector& opL, const uint64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_sra(opL[i_element], rhs);
 	}
 	return(*this);
@@ -583,7 +583,7 @@ SVector& SVector::m_sra(const SVector& opL, const uint64_t rhs, const SVRegister
 
 SVector& SVector::m_srl(const SVector& opL, const SVector& rhs,  const SVRegister& vm, bool mask, size_t start_index) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_srl(opL[i_element], rhs[i_element]);
 	}
 	return(*this);
@@ -591,7 +591,7 @@ SVector& SVector::m_srl(const SVector& opL, const SVector& rhs,  const SVRegiste
 
 SVector& SVector::m_srl(const SVector& opL, const uint64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element].s_srl(opL[i_element], rhs);
 	}
 	return(*this);
@@ -600,7 +600,7 @@ SVector& SVector::m_srl(const SVector& opL, const uint64_t rhs, const SVRegister
 SVector& SVector::m_slideup(const SVector& opL, const uint64_t rhs, const SVRegister& vm, bool mask, size_t start_index ) {
 	size_t max = rhs > start_index ? rhs : start_index;
 	for(size_t i_element = max; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element))
+		if(!mask || vm.get_bit(i_element))
 			(*this)[i_element] = opL[i_element];
 	}
 	return(*this);
@@ -608,7 +608,7 @@ SVector& SVector::m_slideup(const SVector& opL, const uint64_t rhs, const SVRegi
 
 SVector& SVector::m_slidedown(const SVector& opL, const uint64_t rhs, const SVRegister& vm, bool mask, size_t vlmax, size_t start_index ) {
 	for(size_t i_element = start_index; i_element < length_; ++i_element) {
-		if(!mask or vm.get_bit(i_element)) {
+		if(!mask || vm.get_bit(i_element)) {
 			size_t i_src_element = i_element + rhs;
 			if(i_src_element < length_) {
 				(*this)[i_element] = opL[i_element + rhs];

